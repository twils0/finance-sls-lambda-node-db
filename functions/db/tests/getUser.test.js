const { getUser } = require('../getUser');

global.pool = {
  query: jest.fn(),
};

const userId = 'testUserID';
const email = 'test@test.com';
const dbResponse = {
  rows: [
    {
      email,
    },
  ],
};

describe('functions', () => {
  describe('db', () => {
    describe('getUser', () => {
      afterEach(() => {
        global.pool.query.mockReset();
      });

      it('fails and returns unexpected error, when global.pool.query throws error, when provided userId', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([userId]);
          return Promise.reject(error);
        });

        try {
          await getUser(userId);
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it('correctly returns email when provided userId', async () => {
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([userId]);
          return Promise.resolve(dbResponse);
        });

        const result = await getUser(userId);

        expect(result).toEqual({ username: email });
      });
    });
  });
});
