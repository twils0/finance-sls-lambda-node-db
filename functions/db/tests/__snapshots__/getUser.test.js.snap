// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions db getUser correctly returns email when provided userId 1`] = `
"SELECT email
        FROM \\"user\\".\\"user\\"
        WHERE user_id = $1::uuid;"
`;

exports[`functions db getUser fails and returns unexpected error, when global.pool.query throws error, when provided userId 1`] = `
"SELECT email
        FROM \\"user\\".\\"user\\"
        WHERE user_id = $1::uuid;"
`;
